lib/ansible/modules/known_hosts.py

```diff
@@ -222,7 +222,8 @@ def sanity_check(module, host, key, sshkeygen):
         rc, stdout, stderr = module.run_command(sshkeygen_command)
 
     if stdout == '':  # host not found
-        module.fail_json(msg="Host parameter does not match hashed host field in supplied key")
+        module.fail_json(msg="Host parameter does not match hashed host field in supplied key. "
+                             "For custom SSH port, name needs to specify port as well.")
```

Comment:
This is not the only reason this can happen, I would say it is not even the most common one.

I would redirect to ssh documentation on how the hash is determined vs having every possible modifier suggested here.

Reply:
Right. Then I would just close this PR because referring to the SSH documentation is not that helpful as highlighting a potential cause of the error (checking the documentation is probably the first thing one does when something goes wrong, so no need to point that out).